README.rst
======
README
======


ChangeLog
UNRELEASED

  * Remove duplicate `libfred` tests

2022-10-12 (2.55.0)

  * fred-mifd now uses messenger to send emails/letters/sms

2022-09-22 (2.54.1)

  * Fix registrar epp auth operations in libfred (bump deps)

2022-09-14 (2.54.0)

  * Registry objects authinfo password is now stored hashed and has ttl (expiration timestamp)

2022-07-28 (2.53.1)

  * Invoicing
    - switch from doc2pdf to secretary/libytpist
    - switch from mailer/pyfred to messenger/libhermes
    - switch from filamen/file manager to fileman/libfiled

2022-05-19 (2.52.0)

  * Public requests - switch from mailer/pyfred and fileman to messenger and libfiled

2022-04-29 (2.51.0)

  * Add --send_notifications option to fred-admin --object_regular_procedure call. Default is not to send email notifications.

2022-02-28 (2.50.0)

  * Epp - generate update poll messages in all update operations when not done by sponsoring registrar

  * Domain browser - normalize string data of contact (empty string / null values / whitespace) when finding merge candidates

  * Rework object factories to manual registration instead of auto-registration which was not working properly

  * Fix unix whois registrar search (upper case registrar handle)

2022-02-01 (2.49.0)

  * Add possibility to mark registrar as internal
    - not in registrar list in public interfaces but still traceable when linked to another object
    - not charged

  * Epp - on succesfull info_contact command used with authinfo parameter (to get all contact private data)
          generate new authinfo password (same behaviour as transfer_* commands)

  * Epp / Public request - allow to send contact authinfo on registry e-mail even when contact has serverTransferProhibited status

  * Update CMake build

2021-12-22 (2.48.2)

  * Fix mojeid - add specific exception (instead of internal server error) when transferring contact attached to external identity

  * Fix domain browser contact merge candidate list - exclude contacts attached to external identity

  * Fix rpm build (fedora 35) and remove support for CentOS7

2021-12-14 (2.48.1)

  * Quick fix - auto registration tests in admin. contact verification

2021-11-08 (2.48.0)

  * Add concept of external identity (service) that can be linked to registry contact

  * Modify domain browser backend to be usable with both mojeid contacts and contacts linked to external identity service

  * Add new contact states to inform that some of the contact attributes are locked and cannot be changed
    - appropriate changes in epp contact update / contact auto merge procedure

2021-11-05 (2.47.1)

  * Fix rpm build (fedora 35) and remove support for CentOS7

2021-05-18 (2.47.0)

  * Add configuration for epp backend to specify privacy policy for contact data between registrars

  * Add optional authinfo parameter to info contact method

2021-08-09 (2.46.2)

  * Remove explicit python call in external command running filemanager_client

  * Fix update contact poll message additional recipients (bump libfred)

2021-06-14 (2.46.1)

  * Public request processing - add ``registrar_url`` parameter to epp sendauthinfo e-mail template context

2021-04-21 (2.46.0)

  * Whois backend - Optimize info domain method (bump libfred)

  * MojeID - Add ``auto_pin3_sending`` configuration option

2021-03-18 (2.45.9)

  * Fix invoice list - loading of related objects

2021-02-23 (2.45.8)

  * Fix authinfo sending when not confirmed automatically

2020-12-15 (2.45.7)

  * Fix poll message unittests

  * Fix ``add_credit_to_invoice`` function (when used without outer transaction)

  * Fix authinfo e-mail type when requested through epp

  * Fix CI

2020-11-02 (2.45.6)

  * Fix retrieving registrars blocked "today"

2020-10-20 (2.45.5)

  * Validators as inline functions

2020-10-01 (2.45.4)

  * Fix missing parameter validators

2020-08-13 (2.45.3)

  * Add domain_lifecycle_parameters commands into fred-admin

2020-08-11 (2.45.2)

  * Fix domain filter performance issue

2020-08-07 (2.45.1)

  * Fix domain filter for Daphne

2020-07-13 (2.45.0)

  * Use parameters from new table domain_lifecycle_parameters

2020-02-20 (2.44.0)

  * MojeID
    - add new method for validated data update (no mojeid verification states canceling)
    - fix exception in send_new_pin3 method

2020-02-03 (2.43.0)

  * Add man pages for main `fred-admin` commands

2020-01-31 (2.42.3)

  * Fix rpm for RHEL8 and F31

2020-01-28 (2.42.2)

  * fred-admin - add `--charge-to-end-of-previous-month` option to `charge_registry_access_fee_*` commands

2020-01-08 (2.42.1)

  * Invoice export
    - add administrative fee operation
    - remove generic fine operation

2019-09-11 (2.42.0)

  * fred-admin
    - new commands for charging registrar access fee
      - `charge_registry_access_fee_annual`
      - `charge_registry_access_fee_monthly`
    - fix `registrar_add_zone` command - prevent duplicate records
    - fix `registrar_add` command - unhandled exception on invalid varsymbol

  * Adapt to new random data generator interface

  * Registrar credit manipulation and initialization using new
    libfred `CreateRegistrarCreditTransaction` operation

  * adifd - return possibility to change registrar handle

  * Update spec file for F31 and Centos/RHEL 8

2019-11-27 (2.41.2)

  * Add Fee and Fine operations to invoice export

  * Fix export for account invoices with not paid amount

2019-11-08 (2.41.1)

  * VAT calculation changes

2019-07-18 (2.41.0)

  * Epp - create poll message when contact or domain is deleted administratively (to notify sponsoring registrar)

  * Epp, MojeID - create poll message when contact is changed to:
    - sponsoring registrar of this contact if change was done by other than sponsoring registrar
    - sponsoring registrars of domains where this contact is assigned as holder or admin-c

  * Epp - fix contact data change detection (to avoid dropping contact verification states)

  * MojeID - fix contact address discloseflag change due to loss of contact

conf.h
#ifndef __CONF_H__
#define __CONF_H__

#ifdef XMLCONF
#include <libxml/parser.h>
#include <libxml/tree.h>
#endif

#ifndef LOG_LOCAL0
#include <syslog.h>
#endif

#define KEY_dbname      1
#define KEY_user        2
#define KEY_pass        3  
#define KEY_host        4
#define KEY_port        5
#define KEY_timeout     6
#define KEY_log_mask    7
#define KEY_log_level   8
#define KEY_log_local   9
#define KEY_nameservice 10
#define KEY_session_max  11
#define KEY_session_wait  12
#define KEY_docgen_path  13
#define KEY_fileclient_path  14
#define KEY_ebanka_url  15
#define KEY_docgen_template_path  16
#define KEY_nsset_level  17
#define KEY_restricted_handles  18
#define KEY_disable_epp_notifier  19
#define KEY_session_registrar_max 20
#define KEY_lock_epp_commands  21
#define MAX_KEYS  21

#include <string>

class Conf
{
public:

  Conf()
  {
    // default  values 
    port[0]=0;
    timeout[0] = 0;
    host[0] = 0;
    user[0]=0;
    password[0] = 0;
    host[0]=0;
    log_level = 0;
    conninfo[0] =0;
    log_local=0;
    nssetLevel="3";
    restrictedHandles=1;
    disableEPPNotifier=0;
    session_max=20;
    session_wait=300;
    session_registrar_max=5;
    lockEPPCommands=1;
  }
  ~Conf()
  {
  }
  ; // empty

  bool ReadConfigFile(
    const char *filename)
  {
#ifdef XMLCONF
    return ReadConfigFileXML( filename );
#else
    return ReadConfigFileTXT(filename);
#endif
  }
  ;

#ifdef XMLCONF
  void get_element_names(xmlNode * a_node);
  bool ReadConfigFileXML(const char *filename );
#endif
  bool ReadConfigFileTXT(
    const char *filename);

  int GetSessionMax()
  {
    return session_max;
  }
  ;
  int GetSessionRegistrarMax()
  {
    return session_registrar_max;
  }
  ;
  int GetSessionWait()
  {
    return session_wait;
  }
  ;

  const char *GetDBhost()
  {
    if (host[0] == 0)
      return NULL;
    else
      return host;
  }
  ;
  const char *GetDBname()
  {
    if (dbname[0] == 0)
      return NULL;
    else
      return dbname;
  }
  ;
  const char *GetDBuser()
  {
    if (user[0] == 0)
      return NULL;
    else
      return user;
  }
  ;
  const char *GetDBpass()
  {
    if (password[0] == 0)
      return NULL;
    else
      return password;
  }
  ;
  const char *GetDBport()
  {
    if (port[0] == 0)
      return NULL;
    else
      return port;
  }
  ;
  const char *GetDBtimeout()
  {
    if (timeout[0] == 0)
      return NULL;
    else
      return timeout;
  }
  ;
  const char *GetDBconninfo();
  const char *GetNameServiceHost()
  {
    return nameService.c_str();
  }
  const char *GetDocGenPath()
  {
    return docGenPath.c_str();
  }
  const char *GetDocGenTemplatePath()
  {
    return docGenTemplatePath.c_str();
  }
  const char *GetFileClientPath()
  {
    return fileClientPath.c_str();
  }
  const char *GetEBankaURL()
  {
    return eBankaURL.c_str();
  }
  const char *GetNSSetLevel()
  {
    return nssetLevel.c_str();
  }
  unsigned GetRestrictedHandles()
  {
    return restrictedHandles;
  }
  unsigned GetDisableEPPNotifier()
  {
    return disableEPPNotifier;
  }

  int GetSYSLOGlevel()
  {
    return log_level;
  }
  ;
  int GetSYSLOGlocal()
  {
    return log_local;
  }
  ;
  int GetSYSLOGfacility()
  {
    switch (log_local) {
      case 0:
        return LOG_LOCAL0;
      case 1:
        return LOG_LOCAL1;
      case 2:
        return LOG_LOCAL2;
      case 3:
        return LOG_LOCAL3;
      case 4:
        return LOG_LOCAL4;
      case 5:
        return LOG_LOCAL5;
      case 6:
        return LOG_LOCAL6;
      case 7:
        return LOG_LOCAL7;
      default:
        return LOG_LOCAL0;
    }

  }
  ;
  unsigned GetLockEPPCommands()
  {
    return lockEPPCommands;
  }
  int GetLocal(
    char *value);
  int GetLevel(
    char *value);

private:
  char dbname[32];
  char user[32];
  char password[32];
  char host[64];
  char port[16];
  char timeout[10];
  char conninfo[256];
  int log_level;
  int log_local;
  int session_max;
  int session_wait;
  int session_registrar_max;
  std::string nameService;
  std::string docGenPath;
  std::string docGenTemplatePath;
  std::string fileClientPath;
  std::string eBankaURL;
  std::string nssetLevel;
  unsigned restrictedHandles;
  unsigned disableEPPNotifier;
  unsigned lockEPPCommands;
};

#endif
